bee-protocol/src/worker/transaction/processor.rs

```diff
@@ -145,13 +148,25 @@ impl ProcessorWorker {
         if let Some(transaction) = tangle().insert(transaction, hash, metadata) {
             Protocol::get().metrics.new_transactions_inc();
 
-            if !tangle().is_synced() && Protocol::get().requested_transactions.is_empty() {
-                Protocol::trigger_milestone_solidification();
-            }
-
             match Protocol::get().requested_transactions.remove(&hash) {
-                Some((hash, (index, _))) => {
-                    Protocol::trigger_transaction_solidification(hash, index);
+                Some((_hash, (index, _))) => {
+                    let trunk = transaction.trunk();
+                    let branch = transaction.branch();
+
+                    if !tangle().contains(trunk)
+                        && !tangle().is_solid_entry_point(trunk)
+                        && !Protocol::get().requested_transactions.contains_key(trunk)
+                    {
+                        Protocol::request_transaction(*trunk, index);
+                    }
+                    if trunk != branch {
+                        if !tangle().contains(branch)
+                            && !tangle().is_solid_entry_point(branch)
+                            && !Protocol::get().requested_transactions.contains_key(branch)
+                        {
+                            Protocol::request_transaction(*branch, index);
```

Comment:
Actually I think I would put the contain/solid/requested checks in `request_transaction `